app/src/main/java/to/holepunch/bare/android/utils/TelecomUtils.kt

```diff
@@ -0,0 +1,64 @@
+package to.holepunch.bare.android.utils
+
+import android.content.ComponentName
+import android.content.Context
+import android.content.Intent
+import android.telecom.PhoneAccount
+import android.telecom.PhoneAccountHandle
+import android.telecom.TelecomManager
+import android.util.Log
+import to.holepunch.bare.android.MyConnectionService
+
+class PhoneAccountManager(private val context: Context) {
+    private val telecomManager: TelecomManager by lazy {
+        context.getSystemService(Context.TELECOM_SERVICE) as TelecomManager
+    }
+
+    fun getPhoneAccountHandle(): PhoneAccountHandle {
+        return PhoneAccountHandle(
+            ComponentName(context, MyConnectionService::class.java),
+            context.packageName
+        )
+    }
+
+    fun registerPhoneAccount() {
+        try {
+            val phoneAccount = PhoneAccount.builder(getPhoneAccountHandle(), "Bare Android")
+                .setCapabilities(PhoneAccount.CAPABILITY_CALL_PROVIDER)
+                .build()
+            telecomManager.registerPhoneAccount(phoneAccount)
+        } catch (e: Exception) {
+            Log.v("PhoneAccountManager", "Not able to register", e)
+        }
+    }
+
+    fun isPhoneAccountEnabled(): Boolean {
+        return try {
+            telecomManager.getPhoneAccount(getPhoneAccountHandle())?.isEnabled == true
+        } catch (e: SecurityException) {
+            Log.w("PhoneAccountManager", "Missing READ_PHONE_NUMBERS permission", e)
+            false
+        }
+    }
+
+    fun getEnablePhoneAccountIntent(): Intent {
+        return Intent().apply {
+            flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_MULTIPLE_TASK
+            component = ComponentName(
+                "com.android.server.telecom",
+                "com.android.server.telecom.settings.EnableAccountPreferenceActivity"
+            )
+        }
```

Comment:
This is my main concern as it require a manual action from the user.

This is due to the fact that we use

```kt
.setCapabilities(PhoneAccount.CAPABILITY_CALL_PROVIDER
```

But according to the doc, it seems that it is the thing we want

> Can make phone calls (VoIP or otherwise) and want those calls to be integrated into the built-in phone app. Referred to as a system managed [ConnectionService](https://developer.android.com/reference/android/telecom/ConnectionService).

Reply:
I'll prolly ask for guidance on SO.

Reply:
cf https://stackoverflow.com/questions/79542106/voip-on-android-system-manager-or-self-managed